### PR TITLE
VP-5103: Add API with mock data for getting subitems of configurable product

### PR DESCRIPTION
**Pull request by asvishnyakov**

### CODE REVIEW

File: VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public class ApiDemoCatalogController: StorefrontControllerBase
+    {
+        private readonly IDemoCatalogService _demoCatalogService;
+
+        public ApiDemoCatalogController(IWorkContextAccessor workContextAccessor, IStorefrontUrlBuilder urlBuilder, IDemoCatalogService demoCatalogService) : base(workContextAccessor, urlBuilder)
+        {
+            _demoCatalogService = demoCatalogService;
+        }
+
+        // storefrontapi/catalog/search
```
Comment (pushnitsa) [inline VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs]: Remove this comment

File: VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        [HttpGet("catalog/{productId}/configuration")]
+        public async Task<ActionResult<ProductPart[]>> SearchProducts(string productId)
```
Comment (pushnitsa) [inline VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs]: I think we should call this method something like GetProductConfiguration

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs (hunk 2 above)
Comment (asvishnyakov) [reply VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs]: Fixed

File: VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs (hunk 1 above)
Comment (asvishnyakov) [reply VirtoCommerce.Storefront/Controllers/Api/ApiDemoCatalogController.cs]: Fixed

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**